<launch>
	<arg name="ugv_id" default="1"/>
	<arg name="yaw_ref" default="0.0"/>

	<!-- 启动全局规划算法 -->
	<node pkg="sunray_ugv_planning" type="ugv_planing_node" name="ugv_planing_node" output="screen">
		<!-- 无人车编号 -->
		<param name="ugv_planning/ugv_id" value="$(arg ugv_id)" />
		<!-- 无人车高度 -->
		<param name="ugv_planning/ugv_height" value="0.0" />
		<!-- A星算法 重规划频率 -->
		<param name="ugv_planning/replan_time" value="0.2" type="double"/>
		<!-- 路径点发布频率 -->
		<param name="ugv_planning/path_point_pub_frequency" value="0.1" type="double"/>
		<!-- 0 for 全局点云，1 for 局部点云，2 for laserscan, 3代表viobot  -->
		<param name="ugv_planning/map_input_source" value="3" type="int"/>
		<remap from="/ugv$(arg ugv_id)/prometheus/case2/global_pcl" to="/map_generator/global_cloud" />
		<remap from="/ugv$(arg ugv_id)/prometheus/case2/local_pcl" to="/ugv$(arg ugv_id)/map_generator/local_cloud" />
		<remap from="/ugv$(arg ugv_id)/prometheus/case2/laser_scan" to="/ugv$(arg ugv_id)/sensor/2Dlidar_scan" />

		<!-- 地图参数 -->
		<param name="map/origin_x" value="-5.0" type="double"/>
		<param name="map/origin_y" value="-2.5" type="double"/>
		<param name="map/map_size_x" value="10.0" type="double"/>
		<param name="map/map_size_y" value="5.0" type="double"/>
		<param name="map/queue_size" value="1" type="int"/>
		<param name="map/border" value="true" type="bool"/>
		<param name="map/resolution" value="0.1" type="double"/>
		<param name="map/inflate" value="0.2" type="double"/>
		<param name="map/odom_inflate" value="0.3" type="double"/>
		<param name="map/cost_inflate" value="2" type="int"/>
		<param name="astar/lambda_cost" value="10.0" type="double"/>
	</node>

    <!-- Rviz config -->
    <arg name="rviz_enable" default="true"/>
    <arg name="rivz_config" default="$(find sunray_ugv_planning)/launch/viobot.rviz"/>
	<!-- 启动Rviz-->
	<group if="$(arg rviz_enable)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(arg rivz_config)"/>
    </group>
</launch>

